Model/RecommendManagement.php

```diff
@@ -0,0 +1,108 @@
+<?php
+declare(strict_types=1);
+
+namespace Algolia\AlgoliaSearch\Model;
+
+use Algolia\AlgoliaSearch\Api\RecommendClient;
+use Algolia\AlgoliaSearch\Api\RecommendManagementInterface;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Service\IndexNameFetcher;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+class RecommendManagement implements RecommendManagementInterface
+{
+    /**
+     * @var null|RecommendClient
+     */
+    private ?RecommendClient $client = null;
```

Comment:
Let's switch to `protected` to allow developers to customize this implementation via preference if they need.

```suggestion
    protected ?RecommendClient $client = null;
```